syntax = "proto3";
package com.panelsw.ca;
import "errorHandling.proto";
import "obfuscatedString.proto";
option optimize_for = LITE_RUNTIME;

message ConsoleOuputRemap { 

	ObfuscatedString regex = 1;

	bool onMatch = 2;
	ErrorHandling errorHandling = 3;
	bytes promptText = 4;
}

message ExecOnDetails{

	ObfuscatedString command = 1;
	bytes workingdirectory = 2;
	
	map<int32, int32> exitCodeRemap = 3;
	map<string, ObfuscatedString> environment = 4;

	bool async = 5;
	ErrorHandling errorHandling = 6;

	repeated ConsoleOuputRemap consoleOuputRemap = 7;

	// Impersonate another user
	bytes domain = 9;
	bytes user = 10;
	bytes password = 11;

	// Impersonate current user
	bool impersonate = 12;
	uint32 sessionId = 13;
}
